File: app/pets/pets_controller.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.pets.pets_schemas import CreatePetDto, Pet, PetList, UpdatePetDto
from app.pets.pets_service import pets_service
from app.shared.database import get_db

router = APIRouter(
    prefix="/api/students/{studentId}/pets",
    tags=["Pets"],
)


@router.get("")
def find_all(studentId: str, db: Session = Depends(get_db)) -> PetList:
    return pets_service.find_all_for_student(db, studentId)


@router.post("", status_code=201)
def create(studentId: str, body: CreatePetDto, db: Session = Depends(get_db)) -> Pet:
    return pets_service.create(db, studentId, body)


@router.patch("/{petId}")
def update(
    studentId: str, petId: str, body: UpdatePetDto, db: Session = Depends(get_db)
) -> Pet:
    return pets_service.update(db, studentId, petId, body)


@router.delete("/{petId}")
def delete(studentId: str, petId: str, db: Session = Depends(get_db)) -> Pet:
    return pets_service.delete(db, studentId, petId)

File: src/pets/pets.controller.ts
import { Body, Controller, Delete, Get, Param, Patch, Post } from "@nestjs/common"

import { PetsService } from "@/pets/pets.service"
import { CreatePetDto, UpdatePetDto } from "@/pets/pets.dtos"

@Controller("api/students/:studentId/pets")
export class PetsController {
	constructor(private readonly petsService: PetsService) {}

	@Get()
	public async findAll(@Param("studentId") studentId: string) {
		const pets = await this.petsService.findAllForStudent(studentId)
		return {
			total: pets.length,
			items: pets,
		}
	}

	@Post()
	public create(@Param("studentId") studentId: string, @Body() body: CreatePetDto) {
		return this.petsService.create(studentId, body)
	}

	@Patch(":petId")
	public update(
		@Param("studentId") studentId: string,
		@Param("petId") petId: string,
		@Body() body: UpdatePetDto,
	) {
		return this.petsService.update(studentId, petId, body)
	}

	@Delete(":petId")
	public delete(@Param("studentId") studentId: string, @Param("petId") petId: string) {
		return this.petsService.delete(studentId, petId)
	}
}
